Create Auto Fishing - Estate.xml

The-Synthax opened the pull request:
(See previous PR: #33)

Fixes #29

The-Synthax committed:
Create Auto Fishing - Estate.xml

kpreisser commented on TTMouseclickSimulator/SampleProjects/ToontownRewritten/Auto Fishing - Estate.xml:
```diff
@@ -0,0 +1,75 @@
+	  		Use the automatic fishing action for estate 20 times,
+       		and ensure Toon remains fishing instead of going to bed.
+      -->
+      <Loop count="5"> 
```
Is there a reason for only repeating the auto fishing 5 times, then pressing the down/up keys, and then fishing again? If the toon has enough jellybeans for 20 casts, I think this should probably not be needed, and it should be possible to do 19 or 20 casts in a row.

The-Synthax replied:
Probably not, I don't even remember why I did this but it may have had something to do with preventing timeout? If it works without though then that change would make sense.

kpreisser commented on TTMouseclickSimulator/SampleProjects/ToontownRewritten/Auto Fishing - Estate.xml:
```diff
@@ -0,0 +1,75 @@
+       		and ensure Toon remains fishing instead of going to bed.
+      -->
+      <Loop count="5"> 
+        <AutomaticFishing scan1="248, 239" scan2="1492, 660" bubbleColorRgb="49, 128, 120"
```
I noticed that the taget Y position of the fishing lure is too high, probably because the estate fishing uses a different perspective. This can be improved by setting the `scanResultYAdjustment` attribut, for example (I also improved the scan coordinates and colors a bit):
```xml
        <AutomaticFishing scan1="200, 150" scan2="1492, 660" scanResultYAdjustment="35"
                          bubbleColorRgb="56, 129, 122" toleranceRgb="7, 5, 5" />
```

The-Synthax replied:
looks good!

kpreisser commented on TTMouseclickSimulator/SampleProjects/ToontownRewritten/Auto Fishing - Estate.xml:
```diff
@@ -0,0 +1,75 @@
+        <Pause duration="2200" />
+
+      <!-- Allign Toon with board. -->
+      <Compound type="Sequential" minimumPause="50" maximumPause="300" loop="false">
+	    <PressKey key="Down" duration="300" />
+		<PressKey key="Up" duration="1000" />
+		<Pause duration="2200" />
+		<QuitFishing />
+		<Pause duration="2200" />
```
Nice idea to align the toon with the board. My only concern is it might look a bit "strange" if the toon quits fishing, goes on the board again and immediately quits fishing again. Do you mind if we change this to match the pattern of the other files, where the last (20th) cast is a `<StraightFishing />`, to align the toon with the board?

The-Synthax replied:
Yeah that might be a bit less suspicious looking

kpreisser commented on TTMouseclickSimulator/SampleProjects/ToontownRewritten/Auto Fishing - Estate.xml:
```diff
@@ -0,0 +1,75 @@
+		<Pause duration="2200" />
+
+      <!-- Now go to the fisherman and sell the fish. -->
+        <PressKey key="Left" duration="1200" />
```
In my tests I had to increase this duration to `1300` to correctly position the toon (maybe TTR changed the movement speed in recent years):
```suggestion
        <PressKey key="Left" duration="1300" />
```

kpreisser commented on TTMouseclickSimulator/SampleProjects/ToontownRewritten/Auto Fishing - Estate.xml:
```diff
@@ -0,0 +1,75 @@
+
+        <Pause duration="2000" />
+        <PressKey key="Down" duration="2500" />
+        <Pause duration="2000" />
```
The pause duration can be increased a bit:
```suggestion
        <Pause duration="3200" />
```

kpreisser reviewed (commented):
Thank you for this PR, it looks great! I have some comments about the individual steps.

Are you okay with me editing the file contents based on my comments?
Thanks!

The-Synthax commented:
> Thank you for this PR, it looks great! I have some comments about the individual steps.
> 
> Are you okay with me editing the file contents based on my comments? Thanks!

Yes totally, go ahead. I haven't played in quite some time and don't entirely recall my reasoning for certain things here.

kpreisser committed:
Minor improvements to estate fishing:
- Use a loop with count 19, and then a straight cast (to align the toon with the fisherman).
- Improve key press durations and pause times.
- Improve scan color values.

kpreisser reviewed (approved):
Thanks for your contribution! I applied the proposed changes and will merge the PR.
